### ISSUE TITLE
Add the card count to the deck's sidebar

### ISSUE THREAD

**Issue by Ajordat** [Ajordat/alteredbuilder#28]
When building a deck from the card list view, it'd be better to show the total count of the cards in the deck

**Comment by mariags803** [comment]
Hi! I would like to work in this enhancement.

**Comment by Ajordat** [comment]
Hi Maria!

Many thanks for looking into this. First try to replicate the dev environment by following the instructions on the [README.md](https://github.com/Ajordat/alteredbuilder/blob/master/README.md#development).

Once that is done you'll be able to work on this feature.

**Comment by Ajordat** [comment]
The idea would be to add the total values on top or below the current card count:

![image](https://github.com/user-attachments/assets/cc49f438-269b-4420-9d3f-3aeed0e7ec6e)

I think something simple like the following should be enough:
```
3 cards
<line>
1 common 1 rare 1 unique
```

**Comment by Ajordat** [comment]
That's odd, the base images have always worked and they're the same in dev and prod. What error did you encounter?

**Comment by mariags803** [comment]
This is the error I was getting:

```
14.07 Collecting grpcio~=1.62.0 (from -r requirements.txt (line 33))
14.10   Downloading grpcio-1.62.3.tar.gz (26.3 MB)
15.02      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 26.3/26.3 MB 28.7 MB/s eta 0:00:00
16.86   Installing build dependencies: started
19.12   Installing build dependencies: finished with status 'done'
19.13   Getting requirements to build wheel: started
19.60   Getting requirements to build wheel: finished with status 'error'
19.61   error: subprocess-exited-with-error
19.61   
19.61   × Getting requirements to build wheel did not run successfully.
19.61   │ exit code: 1
19.61   ╰─> [26 lines of output]
19.61       <string>:40: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
19.61       Traceback (most recent call last):
19.61         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
19.61           main()
19.61         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
19.61           json_out['return_val'] = hook(**hook_input['kwargs'])
19.61                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
19.61         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
19.61           return hook(config_settings)
19.61                  ^^^^^^^^^^^^^^^^^^^^^
19.61         File "/tmp/pip-build-env-df2yisqu/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 332, in get_requires_for_build_wheel
19.61           return self._get_build_requires(config_settings, requirements=[])
19.61                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
19.61         File "/tmp/pip-build-env-df2yisqu/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 302, in _get_build_requires
19.61           self.run_setup()
19.61         File "/tmp/pip-build-env-df2yisqu/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 503, in run_setup
19.61           super().run_setup(setup_script=setup_script)
19.61         File "/tmp/pip-build-env-df2yisqu/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 318, in run_setup
19.61           exec(code, locals())
19.61         File "<string>", line 273, in <module>
19.61         File "<string>", line 217, in check_linker_need_libatomic
19.61         File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__
19.61           self._execute_child(args, executable, preexec_fn, close_fds,
19.61         File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child
19.61           raise child_exception_type(errno_num, err_msg, err_filename)
19.61       FileNotFoundError: [Errno 2] No such file or directory: 'c++'
19.61       [end of output]
19.61   
19.61   note: This error originates from a subprocess, and is likely not a problem with pip.
19.80 error: subprocess-exited-with-error
19.80 
19.80 × Getting requirements to build wheel did not run successfully.
19.80 │ exit code: 1
19.80 ╰─> See above for output.
19.80 
19.80 note: This error originates from a subprocess, and is likely not a problem with pip.
------
failed to solve: process "/bin/sh -c pip3 install -r requirements.txt --no-cache-dir" did not complete successfully: exit code: 1
```

**Comment by mariags803** [comment]
Hi! I have added the card counter and I would like to know if this was your idea.

![Captura de pantalla 2024-10-11 a las 13 55 59 p  m](https://github.com/user-attachments/assets/ae7ae78e-22fa-4fed-8ffb-d5fe08c1b1e5)

Anything you would like to change, let me know.

From what I saw in the README, the translation part is done after uploading the changes, right?

Thank you very much!

**Comment by Ajordat** [comment]
Hi Maria!

Yes indeed, this is what I had in mind! It looks great!

As per the translations, TLDR: No need to stress about it, but you can already add any translations that you know.

The general workflow works like this:

1. Generate the strings with `makemessages`
2. When pushed to master, Weblate will notice those changes and add those strings to its platform.
3. Once translated, Weblate will push the changes to the translations branch and start a PR.
4. I will merge those changes and I will run `compilemessages` to generate the actual translated strings.

That said, after step 1, if you know any of the other languages used, you can go to the `.po` file and add the translation. On step 2, it will appear as translated on Weblate already.

**Comment by Ajordat** [comment]
About the base image, I'm not sure what's going on actually I've never experienced that. Maybe in the definition I'm not using a narrow-enough version and it got updated? I'm just theorizing.

Btw, feel free to create a PR with your changes! I'll be back home on Tuesday so I'll be able to take a look at it after that (hopefully sooner!)

**Comment by Ajordat** [comment]
Hi María,

I'm finally home so I can finally resume work on the platform. Create the PR whenever you're ready and I'll take a look.

**Comment by mariags803** [comment]
Thank you very much!
I will create the PR for the docker image issue. 
I want to review the card counter code and create the translations before I create that PR. I hope to have it ready in these days.

**Comment by mariags803** [comment]
Hi, how are you?

I have a question, may I add new tests to this feature?

**Comment by Ajordat** [comment]
Hi Maria, absolutely. The reason there aren't so far for frontend technologies is simply that _I don't know_ how to do it 🤦‍♂️

So please, do not hesitate to add tests for this frontend feature if you can.

**Comment by Ajordat** [comment]
Hi María, how are you doing with this feature?

I'm currently working on a redesign of  the whole page and, whilst I'm modifying certain behaviors of the deckbuilder, I'd like to merge your changes regardless.

Let me know!

**Comment by mariags803** [comment]
Hello! Sorry for the delay. I've been busy these days. I would like to create the PR between today and tomorrow.

Thank you!

**Comment by mariags803** [comment]
I have created the PR but without adding the tests because I don't understand very well how to launch them from Docker, as it is done for Python tests, sorry.

**Comment by Ajordat** [comment]
No problem! I've started to take a look at your PR but I'll resume it within tomorrow and Friday.

Many thanks for your efforts! I'll most likely deploy it as soon as I have it reviewed as it seemed alright at first glance 😄

**Comment by Ajordat** [comment]
I just merged your PR into master! 🎉 

I'd love to credit you on the [collaborators page](https://altered.ajordat.com/en/collaborators/). Is that okay with you? Do you want me to link to your Github account or shall I reference you only by "mariags803"?

**Comment by mariags803** [comment]
Great! So glad I could help! 🎉 😄

Yeah, sure, you can link to my Github no problem.

Thanks a lot! I'd like to continue to help where I can if possible. 😊

**Comment by Ajordat** [comment]
Awesome! I just added you to the collaborators page then! (4aa306d5c6ee6917ca34b34c4362c65a1ebc441f)

It'd be awesome if you could keep on contributing! Do you have a preference on what would you like to work on? Frontend or backend stuff?

In the same page that you worked on there are a couple of improvements that I wanted to do:
* Add a view button to go directly to the deck's page. As I think about it, most likely I'd add it to the right of the save button.
  * This is purely frontend stuff and it's a matter of adding a single anchor.
* Add a switch to make the deck private or public easily. By default it should still be public, but it could be helpful or people to have easier access to that feature.
  * This would require adding the parameter on the request that saves the changes of the deck (save button) and then read it on the Python backend to modify the database.
* Add a floating scroll button to go back to the top of the page when the user starts scrolling a few rows of cards.
  * This is purely frontend stuff and it's a matter of adding some JS to make it appear after a certain height and to make it go up to the page top.
* Improve how responsive the sidebar is. With the latest redesign I changed the sidebar to a div element instead, however I don't think it's as responsive as it should.
  * For example I'd like the name to be cut into an ellipsis depending on the length of its parent container, which is not the case currently. As it works right now, there's a series of breakpoints defined but the name is limited using max-width and a fixed number of pixels which is far from ideal.
  * I can probably develop this more if you're interested in this kind of stuff that is purely frontend technologies.

If you already had an idea or would want to work on something that is described in another issue, feel free to mention it.

I'll mark this issue as closed, but we can continue the conversation here or through Discord.

**Comment by Ajordat** [comment]
Feel free to check the associated project where you could find other ideas that I'll eventually get to: https://github.com/users/Ajordat/projects/1

**Comment by mariags803** [comment]
Thank you very much for adding me!! 🥳

I feel more comfortable with the frontend, because I've barely worked on Django. But I don't mind if I have to modify something on the backend.

I think these are great ideas and I'd like to do the frontend parts if it's ok with you. 😊

If I think of any improvements, I'll let you know and we can continue talking on Discord.

**Comment by Ajordat** [comment]
Awesome! Then feel free to pick your poison 😂  or let me know if you have other suggestions ofc
